src/server/scripts/EasternKingdoms/Scholomance/boss_kirtonos_the_herald.cpp

```diff
@@ -128,6 +118,25 @@ class boss_kirtonos_the_herald : public CreatureScript
             }
         }
 
+        void SetData(uint32 type, uint32 data) override
```

Comment:
In that case, you could also use DoAction()

Reply:
Done, didn't know it existed.